include/libp2p/basic/scheduler.hpp

```diff
@@ -0,0 +1,239 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef LIBP2P_BASIC_SCHEDULER_HPP
+#define LIBP2P_BASIC_SCHEDULER_HPP
+
+#include <chrono>
+#include <functional>
+#include <memory>
+
+#include <libp2p/outcome/outcome.hpp>
+
+namespace libp2p::basic {
+
+  /**
+   * Feedback from scheduler backend to Scheduler implementation.
+   */
+  class SchedulerBackendFeedback {
+   public:
+    virtual ~SchedulerBackendFeedback() = default;
+
+    /**
+     * Called from backend to fire callbacks
+     * @param current_clock
+     * For timers: non-zero current async ticks in milliseconds since async's
+     * epoch;
+     * For callbacks deferred to next IO loop cycle: zero
+     */
+    virtual void pulse(int64_t current_clock) noexcept = 0;
```

Comment:
```suggestion
    virtual void pulse(std::chrono::milliseconds current_clock) noexcept = 0;
```
Can convert from `now()` (time_point) with `.time_since_epoch()`